x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java

```diff
@@ -189,4 +195,42 @@ static DeprecationIssue checkTemplatesWithMultipleTypes(ClusterState state) {
             + " define multiple types and so will cause errors when used in index creation"
             );
     }
+
+    static DeprecationIssue checkAutoFollowedSystemIndices(ClusterState state) {
+        AutoFollowMetadata autoFollowMetadata = state.metadata().custom(AutoFollowMetadata.TYPE);
+        final List<String> followedLeaderIndexUUIDs;
+        if (autoFollowMetadata == null) {
+            followedLeaderIndexUUIDs = Collections.emptyList();
+        } else {
+            followedLeaderIndexUUIDs = autoFollowMetadata.getFollowedLeaderIndexUUIDs()
+                .values()
+                .stream()
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+        }
+
+        Set<String> systemIndexFollowers = new HashSet<>();
+        for (ObjectObjectCursor<String, IndexMetadata> indexEntry : state.metadata().getIndices()) {
+            final IndexMetadata indexMetadata = indexEntry.value;
+            final Map<String, String> ccrMetadata = indexMetadata.getCustomData(CcrConstants.CCR_CUSTOM_METADATA_KEY);
+            if (ccrMetadata == null) {
+                continue;
+            }
+            final String leaderIndexUUID = ccrMetadata.get(CcrConstants.CCR_CUSTOM_METADATA_LEADER_INDEX_UUID_KEY);
+            if (indexMetadata.isSystem() && followedLeaderIndexUUIDs.contains(leaderIndexUUID)) {
```

Comment:
I were a bit surprised that `isSystem()` is true here, but I did not find the place where we would filter that out. I suppose it is correct, since it is sort of a restore. Would be nice to validate though in a test?

Reply:
We don't filter that metadata out.  I'll add a test to validate it 👍